routes/v1.js

```diff
@@ -147,10 +147,25 @@ function addRoutes(app, peliasConfig) {
   );
 
   // defer to pelias parser for analysis IF there's no response AND placeholder should not have executed
-  const shouldDeferToPeliasParser = all(
-    not(predicates.hasRequestErrors),
-    not(predicates.hasResponseData)
-  );
+  const shouldDeferToPeliasParser = function(req, res) {
+    if (not(predicates.hasResponseData)(req, res)) {
+      return true;
+    }
+    const ph = placeholderShouldHaveExecuted(req,res);
```

Comment:
Is there any advantage to returning here if `ph === false` to avoid doing the work below in cases where it'll never influence the result?

Reply:
This code was written more to facilitate my own debugging than anything else. I had to split up the logic and insert lots of `console.log`s to really figure out what was happening.

One the one hand, I like the flexibility. On the other hand it probably shouldn't stay in the final version.

Reply:
To be clear, I'm hoping that before merging this PR we can make the logic clear and clean enough that this function can be defined more similarly to the other predicate functions, where it's composed purely of other predicates without having to define a Javascript function directly.

Reply:
:+1: sorry I'll stop reviewing your draft PR 😝

Reply:
Its ready for real review now, check it out :)